src/CoreBundle/Response/ResponseVariableReplacerInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\Response;
+
+use ChameleonSystem\CoreBundle\Security\AuthenticityToken\TokenInjectionFailedException;
+
+interface ResponseVariableReplacerInterface
+{
+    /**
+     * Add a variable that can then be replaced by replaceVariables(). Note that variables MUST alreday be escaped by
+     * the caller, as implementations of this interface do not know the context in which the variables are used and are
+     * therefore unable to perform the correct escaping.
+     *
+     * @param string $key
+     * @param string $value
+     */
+    public function addVariable(string $key, string $value): void;
```

Comment:
I do not have an easy solution for this but with this interface we force a state which leads to a stateful service if used as such by an implementation.

Reply:
See my comment above - I think this is OK (and suppose we can't get around state here anyway).